File: androidApp/src/main/kotlin/com/example/app/android/components/profile/ProfileHeader.kt
package com.example.app.android.components.profile

import android.graphics.BitmapFactory
import android.net.Uri
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.asImageBitmap
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.app.android.theme.AppTheme

@Composable
fun ProfileHeader(
    name: String,
    email: String,
    initials: String,
    photoUri: Uri?,
    onAvatarClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val colors = AppTheme.colors
    val context = LocalContext.current

    val photoBitmap = remember(photoUri) {
        photoUri?.let { uri ->
            try {
                context.contentResolver.openInputStream(uri)?.use { stream ->
                    BitmapFactory.decodeStream(stream)?.asImageBitmap()
                }
            } catch (_: Exception) {
                null
            }
        }
    }

    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Box(
            modifier = Modifier.clickable(onClick = onAvatarClick),
            contentAlignment = Alignment.Center
        ) {
            // Avatar
            Box(
                modifier = Modifier
                    .size(80.dp)
                    .clip(CircleShape)
                    .background(colors.accent),
                contentAlignment = Alignment.Center
            ) {
                if (photoBitmap != null) {
                    Image(
                        bitmap = photoBitmap,
                        contentDescription = null,
                        modifier = Modifier.size(80.dp).clip(CircleShape),
                        contentScale = ContentScale.Crop
                    )
                } else {
                    Text(
                        text = initials,
                        fontSize = 28.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = colors.onAccent
                    )
                }
            }

            // Camera badge
            Box(
                modifier = Modifier
                    .align(Alignment.BottomEnd)
                    .offset(x = 2.dp, y = 2.dp)
                    .size(28.dp)
                    .clip(CircleShape)
                    .background(colors.surface),
                contentAlignment = Alignment.Center
            ) {
                Box(
                    modifier = Modifier
                        .size(24.dp)
                        .clip(CircleShape)
                        .background(colors.accent),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.CameraAlt,
                        contentDescription = null,
                        tint = colors.onAccent,
                        modifier = Modifier.size(14.dp)
                    )
                }
            }
        }

        Spacer(modifier = Modifier.height(12.dp))

        Text(
            text = name,
            fontSize = 20.sp,
            fontWeight = FontWeight.SemiBold,
            color = colors.textPrimary
        )

        Spacer(modifier = Modifier.height(2.dp))

        Text(
            text = email,
            fontSize = 14.sp,
            color = colors.textSecondary
        )
    }
}

File: iosApp/iosApp/Components/Profile/ProfileHeader.swift
import SwiftUI
import UIKit

struct ProfileHeader: View {
    let name: String
    let email: String
    let initials: String
    var photoData: Data?
    var onAvatarTap: () -> Void = {}

    @Environment(\.appColors) private var colors

    var body: some View {
        VStack(spacing: 0) {
            // Avatar with camera badge
            Button(action: onAvatarTap) {
                ZStack(alignment: .bottomTrailing) {
                    // Avatar circle
                    ZStack {
                        Circle()
                            .fill(colors.accent)
                            .frame(width: 80, height: 80)

                        if let data = photoData, let uiImage = UIImage(data: data) {
                            Image(uiImage: uiImage)
                                .resizable()
                                .scaledToFill()
                                .frame(width: 80, height: 80)
                                .clipShape(Circle())
                        } else {
                            Text(initials)
                                .font(.system(size: 28, weight: .semibold))
                                .foregroundColor(colors.onAccent)
                        }
                    }

                    // Camera badge
                    ZStack {
                        Circle()
                            .fill(colors.surface)
                            .frame(width: 28, height: 28)

                        Circle()
                            .fill(colors.accent)
                            .frame(width: 24, height: 24)

                        Image(systemName: "camera.fill")
                            .font(.system(size: 11))
                            .foregroundColor(colors.onAccent)
                    }
                    .offset(x: 2, y: 2)
                }
            }
            .buttonStyle(.plain)

            Spacer().frame(height: 12)

            Text(name)
                .font(.system(size: 20, weight: .semibold))
                .foregroundColor(colors.textPrimary)

            Spacer().frame(height: 2)

            Text(email)
                .font(.system(size: 14))
                .foregroundColor(colors.textSecondary)
        }
    }
}
